dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -149,6 +149,35 @@ def delete_progress_form
       count
   end
 
+  def delete_progress
+    params.require([:user_id, :script_id, :reason])
+
+    # TODO: Check for escaping issues for 'reason' once everything is hooked up end-to-end.
+
+    user_id = params[:user_id]
+    script_id = params[:script_id]
+    user_storage_id = storage_id_for_user_id(user_id)
+
+    FirehoseClient.instance.put_record(
+      :analysis,
+      {
+        study: 'reset-progress',
+        event: 'admin-delete-progress',
+        user_id: user_id,
+        script_id: script_id,
+        data_json: {
+          signed_in_user: current_user.username,
+          reason: params[:reason]
+        }.to_json
+      }
+    )
+
+    UserScript.where(user_id: user_id, script_id: script_id).destroy_all
+    UserLevel.where(user_id: user_id, script_id: script_id).destroy_all
+    ChannelToken.where(storage_id: user_storage_id, script_id: script_id).destroy_all unless user_storage_id.nil?
```

Comment:
I'm curious what the case is where user_storage_id is nil.

Last week we were talking about potentially not backfilling script_id on the channel tokens table. This makes a strong case that we do need to backfill (even for those levels where they're only present in one script).

Reply:
`user_storage_id` would be nil if the user never created a channel token.  One possible scenario is we're resetting the progress for a student who has done mostly CSF sources (which mostly store source code in level_sources).